crates/boundless-market/src/contracts/mod.rs

```diff
@@ -430,9 +433,10 @@ impl ProofRequest {
         }
         Url::parse(&self.imageUrl).map(|_| ())?;
 
-        if self.requirements.imageId == B256::default() {
-            return Err(RequestError::ImageIdIsZero);
-        }
+        // TODO(ec2): fixme
```

Comment:
Maybe here we could run a similar check for the predicate, and based on the type check that the corresponding field is not zero